src/service_daemon.rs

```diff
@@ -2356,6 +2356,21 @@ impl DnsCache {
             _ => return None,
         };
 
+        if incoming.get_cache_flush() {
+            // Mark all existing records of this type as expired, and prepend the new record.
+            let now = current_time_millis();
+            record_vec.iter_mut().for_each(|r| {
+                // When cache flush is asked, we set expire date to 1 second in the future
+                // if created more than 1 second ago
+                // Ref: RFC 6762 Section 10.2
+                if now - r.get_created() > 1000 {
+                    r.set_expire(now + 1000);
+                }
+            });
+            record_vec.insert(0, incoming);
```

Comment:
I was wondering the same. Re-reading the RFC it seems like the right way to do this, since we are looking at the records one at a time, and the assumption in the RFC looks to be everything below a second can be part of a burst.